cmd/thanos/store.go

```diff
@@ -49,7 +50,18 @@ func registerStore(m map[string]setupFunc, app *kingpin.Application, name string
 	blockSyncConcurrency := cmd.Flag("block-sync-concurrency", "Number of goroutines to use when syncing blocks from object storage.").
 		Default("20").Int()
 
+	minTime := model.TimeOrDuration(cmd.Flag("min-time", "Start of time range limit to serve. Thanos Store serves only metrics, which happened later than this value. Option can be a constant time in RFC3339 format or time duration relative to current time, such as -1d or 2h45m. Valid duration units are ms, s, m, h, d, w, y.").
```

Comment:
`Default("0000-01-01T00:00:00Z"))` Is that lowest min time? What about `-9999-12-31T23:59:59Z` ?

Reply:
```
    t, err = time.Parse(time.RFC3339, "-9999-12-31T23:59:59Z")
    if err != nil {
        fmt.Println(err)
    }
    fmt.Println(t)
```

gives:
```
parsing time "-9999-12-31T23:59:59Z" as "2006-01-02T15:04:05Z07:00": cannot parse "-9999-12-31T23:59:59Z" as "2006"        
```

Reply:
I would set some explicit default value here (for `""` value), but we can do it in next PRs (: